core/templates/dev/head/player/AnswerClassificationServiceSpec.js

```diff
@@ -0,0 +1,133 @@
+  }));
+
+  var explorationId = 'exploration';
+
+  var state = {
+    interaction: {
+      answer_groups: [{
+        outcome: 'outcome 1',
+        rule_specs: [{
+          inputs: {
+            x: 10
+          },
+          rule_type: 'Equals'
+        }]
+      }, {
+        outcome: 'outcome 2',
+        rule_specs: [{
+          inputs: {
+            x: 5
+          },
+          rule_type: 'Equals'
+        }, {
+          inputs: {
+            x: 7
+          },
+          rule_type: 'NotEquals'
+        }]
+      }],
+      default_outcome: 'default'
+    }
+  };
+
+  var rules = {
+    'Equals': function(answer, inputs) {
+      return inputs.x === answer;
+    },
+    'NotEquals': function(answer, inputs) {
+      return inputs.x != answer;
+    }
+  };
+
+  it('should query the backend when no frontend rules are provided',
+      function() {
+    var backendClassifiedOutcome = {
+      outcome: 'outcome',
+      answer_group_index: 0,
+      rule_spec_index: 0
+    };
+    var expectedClassificationResult = {
+      outcome: 'outcome',
+      answerGroupIndex: 0,
+      ruleSpecIndex: 0
+    };
+    $httpBackend.expectPOST(
+      '/explorehandler/classify/' + explorationId).respond(
+      backendClassifiedOutcome);
+    var classificationResult;
+    acs.getMatchingClassificationResult(explorationId, state, 0).then(
+      function(outcome) {
+        classificationResult = outcome;
+    });
+    $httpBackend.flush();
+    expect(classificationResult).toEqual(expectedClassificationResult);
+    $httpBackend.verifyNoOutstandingExpectation();
+    $httpBackend.verifyNoOutstandingRequest();
+  });
+
+  it('should return the first matching answer group and first matching rule' +
+     'spec', function() {
+    acs.getMatchingClassificationResult(explorationId, state, 10, rules).then(
+      function(classificationResult) {
+        expect(classificationResult).toEqual({
+          outcome: 'outcome 1',
+          answerGroupIndex: 0,
+          ruleSpecIndex: 0
+        });
+    });
+    acs.getMatchingClassificationResult(explorationId, state, 5, rules).then(
+      function(classificationResult) {
+        expect(classificationResult).toEqual({
+          outcome: 'outcome 2',
+          answerGroupIndex: 1,
+          ruleSpecIndex: 0
+        });
+    });
+    acs.getMatchingClassificationResult(explorationId, state, 6, rules).then(
+      function(classificationResult) {
+        expect(classificationResult).toEqual({
+          outcome: 'outcome 2',
+          answerGroupIndex: 1,
+          ruleSpecIndex: 1
+        });
+    });
+  });
+
+  it('should return the default rule if no answer group matches', function() {
+    acs.getMatchingClassificationResult(explorationId, state, 7, rules).then(
+      function(classificationResult) {
+        expect(classificationResult).toEqual({
+          outcome: 'default',
+          answerGroupIndex: 2,
+          ruleSpecIndex: 0
+        });
+    });
+  });
```

Comment:
If getMatchingEditorClassificationResult() still exists after the previous comments, I would add tests for that too. (Though I hope we can get rid of it somehow.)